duplex-messaging.md

```diff
@@ -0,0 +1,74 @@
+# SMP agent protocol - duplex communication over SMP protocol
+
+## Table of contents
+
+- [Abstract](#abstract)
+- [SMP agent commands](#smp-agent-commands)
+
+## Abstract
+
+Protocol describes the command that can be sent to the client that communicates with multiple SMP servers. This protocol allows to manage multiple simplex connections organised into groups defining duplex communication channels.
+
+## SMP agent commands
+
+Commands syntax below is provided using [ABNF][1].
+
+Each transmission between the user and SMP agent must have this format/syntax
+(after unwrapping it out of SMP message):
+
+```abnf
+transmission = (userCmd / agentMsg) CRLF
+userCmd = create / alias / subscribe
+          / invite / accept
+          / send / acknowledge
+          / status / info
+          / suspend / delete
+agentMsg = connection / connectionInfo / invitation / message / unsubscribed / ok / error
+
+create = %s"NEW" SP serverUris ; response is "connection" or "error"
+serverUris = serverUri ["," serverUris]
+
+connection = %s"ID" SP cAlias SP recipientQueuesStatus SP senderQueuesStatus SP cId
+alias = %s"NAME" SP cName SP cId ; response is "connection" or "error"
+status = %"STAT" SP cAlias ; response is "connection" or "error"
+cId = encoded
+cAlias = cId / cName ; TODO add cName syntax - it's letters, numbers, "-" and "_"
```

Comment:
why is connection alias needed on protocol level, or at all?

Reply:
so it's convenient to test via telnet - it's almost chat

Reply:
Same as ":", not really needed. You can memorise short aliases but not IDs. So to send message you can `SEND ef :hello` and it would arrive as `MSG ep 5\nhello` - not chat but already usable, sort of.

Reply:
applications won't use it but this SMP agent is v0.1 terminal chat client